test/cql-pytest/test_describe.py

```diff
@@ -0,0 +1,166 @@
+def test_desc_functions(scylla_only, cql, test_keyspace):
+    fn_schema = "(val int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "AS 'return -val'"
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} {fn_body}") as fn:
+        cql.execute(f"USE {test_keyspace}")
+        result = cql.execute("DESC FUNCTIONS")
+        names = extract_names(result)
+
+        assert fn in names
+
+# Test that `DESC FUNCTION {name}` contains create statement for user function
+def test_desc_function(scylla_only, cql, test_keyspace):
+    fn_schema = "(val int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "return -val"
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} AS '{fn_body}'") as fn:
+        result = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}")
+        desc = extract_describe(result)
+
+        assert f"CREATE FUNCTION {test_keyspace}.{fn}" in desc
+        assert fn_schema in desc
+        assert fn_null in desc
+        assert fn_return in desc
+        assert fn_language in desc
+        assert fn_body in desc
+
+# Test that `DESC AGGREGATES` contains newly created user aggregate
+def test_desc_aggregates(scylla_only, cql, test_keyspace):
+    fn_schema = "(val1 int, val2 int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "return val1 + val2"
+
+    cql.execute(f"USE {test_keyspace}")
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} AS '{fn_body}'") as fn:
+        agg_body = f"(int) SFUNC {fn} STYPE int INITCOND 0"
+        with new_aggregate(cql, test_keyspace, agg_body) as agg:
+            result = cql.execute("DESC AGGREGATES")
+            names = extract_names(result)
+
+            assert agg in names
+
+# Test that `DESC AGGREGATE {name}` contains create statement for user aggregate
+def test_desc_aggregate(scylla_only, cql, test_keyspace):
+    fn_schema = "(val1 int, val2 int)"
+    fn_null = "RETURNS NULL ON NULL INPUT"
+    fn_return = "RETURNS int"
+    fn_language = "LANGUAGE lua"
+    fn_body = "return val1 + val2"
+
+    cql.execute(f"USE {test_keyspace}")
+    with new_function(cql, test_keyspace, f"{fn_schema} {fn_null} {fn_return} {fn_language} AS '{fn_body}'") as fn:
+        agg_body = f"(int) SFUNC {fn} STYPE int INITCOND 0"
+        with new_aggregate(cql, test_keyspace, agg_body) as agg:
+            result = cql.execute(f"DESC AGGREGATE {test_keyspace}.{agg}")
+            desc = extract_describe(result)
+
+            assert f"CREATE AGGREGATE {test_keyspace}.{agg}" in desc
+            assert f"SFUNC {fn}" in desc
+            assert "REDUCEFUNC" not in desc
+            assert "FINALFUNC" not in desc
```

Comment:
We should also have a test case which DOES have those reducefunc and finalfunc.
Personally, I would just have an array of bodies, e.g.,
```python
bodies = [  f"(int) SFUNC {fn} STYPE int INITCOND 0",  f"(int) SFUNC {fn} STYPE int REDUCEFUNC zzz INITCOND 0" ]
```
And then you can loop on each of these bodies creating them and then checking the describe has that entire body (again, I think no need to split it).